packages/cli/src/lib/git.ts

```diff
@@ -155,8 +180,34 @@ export async function findBaseCommit(headRef = 'HEAD', cwd?: string): Promise<st
  * @param {string} [cwd] - The working directory to run the command in.
  * @returns {Promise<string>} A promise containing the diff.
  */
-export function getWorkingDiff(cwd?: string): Promise<string> {
-  return execute('git', ['diff', 'HEAD'], { cwd });
+export async function getWorkingDiff(cwd?: string): Promise<string> {
+  const untrackedFiles = await getUntrackedFiles(cwd);
+  const untrackedDiffs: string[] = [];
```

Comment:
We also list project files for the purpose of building the file index. So it could be useful to compare that with this and see if any other changes should be made in either place, or if the code can be shared.

Reply:
Also, when listing files for search, we detect and ignore binary files and also look at how big they are.

Reply:
It looks like these are also not considered by search. I can create an issue.

In this instance, binary files aren't much of a consideration as they won't render in the diff view, but there's still potential for a large diff to overflow the token limit. In that case, it'd fall back to the default welcome message.